File: app/src/main/java/agersant/polaris/api/IPolarisAPI.java
package agersant.polaris.api;

import com.google.android.exoplayer2.source.MediaSource;

import java.io.IOException;

import agersant.polaris.CollectionItem;

public interface IPolarisAPI {

    MediaSource getAudio(CollectionItem item) throws IOException;

    void browse(String path, final ItemsCallback handlers);

    void flatten(String path, final ItemsCallback handlers);

    void search(String query, final ItemsCallback handlers);

    void getPlaylists(PlaylistsCallback handlers);

    void getPlaylist(String name, ItemsCallback handlers);
}

File: app/src/main/java/agersant/polaris/api/IPolarisAPI.kt
package agersant.polaris.api

import agersant.polaris.CollectionItem
import agersant.polaris.Song
import com.google.android.exoplayer2.source.MediaSource

interface IPolarisAPI {
    suspend fun getAudio(item: Song): MediaSource?

    suspend fun browse(path: String): List<CollectionItem>?

    suspend fun flatten(path: String): List<Song>?
}
